testsuite/integration-arquillian/tests/base/src/test/java/org/keycloak/testsuite/forms/MultipleTabsLoginTest.java

```diff
@@ -368,4 +373,53 @@ public void loginWithSameClientDifferentStatesLoginInTab2() throws Exception {
         assertCurrentUrlStartsWith(redirectUri2);
         Assert.assertTrue(currentUrl.contains("state2"));
     }
+
+    // KEYCLOAK-12161
+    @Test
+    public void testEmptyBaseUrl() throws Exception {
+        String clientUuid = KeycloakModelUtils.generateId();
+        ClientRepresentation emptyBaseclient = ClientBuilder.create()
+                .clientId("empty-baseurl-client")
+                .id(clientUuid)
+                .enabled(true)
+                .baseUrl("")
+                .addRedirectUri("http://localhost:8180/auth/realms/master/app/auth/*")
+                .addRedirectUri("https://localhost:8543/auth/realms/master/app/auth/*")
```

Comment:
Just a small nitpick. I think we could just use single `.addRedirectUri("*")`.

Reply:
Ok, I'll fix that.